iceoryx_examples/waitset/ice_waitset_basic.cpp

```diff
@@ -15,79 +15,84 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "iceoryx_posh/popo/subscriber.hpp"
-#include "iceoryx_posh/runtime/posh_runtime.hpp"
-#include "iceoryx_posh/popo/wait_set.hpp"
 #include "iceoryx_posh/popo/user_trigger.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/cxx/optional.hpp"
 #include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
 #include "topic_data.hpp"
 
-#include <iostream>
 #include <atomic>
+#include <iostream>
 
 std::atomic_bool shutdown{false};
-iox::popo::UserTrigger shutdownTrigger;
+iox::popo::WaitSet<>* waitsetPtr = nullptr;
```

Comment:
Please no raw pointers, wrap it in `optional` or `unique_ptr`

Reply:
Rejected, this example should be as simple as possible and a `cxx::optional` would defeat this purpose.

Reply:
I disagree! Most commonly this code will be copy and pasted, so all the examples should accurately follow our own coding standards!